compiler-core/src/format.rs

```diff
@@ -1704,29 +1704,67 @@ impl<'comments> Formatter<'comments> {
     }
 
     fn call_arg<'a>(&mut self, arg: &'a CallArg<UntypedExpr>, arity: usize) -> Document<'a> {
-        match &arg.label {
-            Some(s) => commented(
-                s.to_doc().append(": "),
-                self.pop_comments(arg.location.start),
-            ),
-            None => nil(),
+        self.format_call_arg(arg, expr_call_arg_formatting, |this, value| {
+            this.comma_separated_item(value, arity)
+        })
+    }
+
+    fn format_call_arg<'a, A, F, G>(
+        &mut self,
+        arg: &'a CallArg<A>,
+        figure_formatting: F,
+        format_value: G,
+    ) -> Document<'a>
+    where
+        F: Fn(&'a CallArg<A>) -> CallArgFormatting<'a, A>,
+        G: Fn(&mut Self, &'a A) -> Document<'a>,
+    {
+        match figure_formatting(arg) {
+            CallArgFormatting::Unlabelled(value) => format_value(self, value),
+            CallArgFormatting::Punned(label) => {
+                let comments = self.pop_comments(arg.location.start);
+                let label = label.as_ref().to_doc().append(":");
+                commented(label, comments)
+            }
+            CallArgFormatting::Labelled(label, value) => {
+                let comments = self.pop_comments(arg.location.start);
+                let label = label.as_ref().to_doc().append(": ");
+                let value = format_value(self, value);
+                commented(label, comments).append(value)
+            }
         }
-        .append(self.comma_separated_item(&arg.value, arity))
     }
 
     fn record_update_arg<'a>(&mut self, arg: &'a UntypedRecordUpdateArg) -> Document<'a> {
         let comments = self.pop_comments(arg.location.start);
-        let doc = arg
-            .label
-            .as_str()
-            .to_doc()
-            .append(": ")
-            .append(self.expr(&arg.value));
+        match arg {
+            // Punned argument.
+            _ if arg.is_punned() => commented(arg.label.as_str().to_doc().append(":"), comments),
+
+            // A call arg that wasn't explicitly punned but could be. We want to
+            // enforce a consistent style and only one way to do things so if
+            // something can be punned it should be.
+            UntypedRecordUpdateArg {
+                label,
+                value: UntypedExpr::Var { name, .. },
+                ..
+            } if label == name => commented(arg.label.as_str().to_doc().append(":"), comments),
```

Comment:
I checked other formatters and asked around what people think of this and it is unpopular. Let's let the programmer pick which syntax to use.